transform/defaults.go

```diff
@@ -30,6 +34,20 @@ func init() {
 	defaultValues["services.*.gpus.*"] = deviceRequestDefaults
 }
 
+// RegisterDefaultTransformer registers a custom transformer for the given path pattern
+func RegisterDefaultTransformer(path string, transformer TransformFunc) {
```

Comment:
`RegisterDefaultTransformer` sounds weird to me. What about `RegisterDefaultValue`